src/dali/abstract_ccxt_input_plugin.py

```diff
@@ -0,0 +1,651 @@
+                #   'addressTo': '0x788cabe9236ce061e5a892e1a59395a81fc8d62c',
+                #   'addressFrom': None,
+                #   'tag': None,
+                #   'tagTo': None,
+                #   'tagFrom': None,
+                #   'type': 'withdrawal',
+                #   'amount': 0.00999800,
+                #   'currency': 'PAXG',
+                #   'status': 'ok',
+                #   'updated': None,
+                #   'internal': False,
+                #   'fee': None
+                # }
+                pagination_detail_iterator.update_fetched_elements(withdrawals)
+
+                with ThreadPool(self._thread_count) as pool:
+                    processing_result_list: List[Optional[ProcessOperationResult]] = pool.map(self._process_transfer, withdrawals)  # type: ignore
+
+                for processing_result in processing_result_list:
+                    if processing_result is None:
+                        continue
+                    if processing_result.intra_transactions:
+                        intra_transactions.extend(processing_result.intra_transactions)
+
+        except StopIteration:
+            # End of pagination details
+            pass
+
+    ### Single Transaction Processing
+
+    def _process_buy_and_sell(self, transaction: Any, notes: Optional[str] = None) -> ProcessOperationResult:
+        results: ProcessOperationResult = self._process_buy(transaction, notes)
+        results.out_transactions.extend(self._process_sell(transaction, notes).out_transactions)
+
+        return results
+
+    def _process_buy(self, transaction: Any, notes: Optional[str] = None) -> ProcessOperationResult:
+        self._logger.debug("Buy: %s", json.dumps(transaction))
+        in_transaction_list: List[InTransaction] = []
+        out_transaction_list: List[OutTransaction] = []
+        crypto_in: RP2Decimal
+        crypto_fee: RP2Decimal
+        fee_asset: str = transaction[_FEE][_CURRENCY]
+
+        trade: Trade = self._to_trade(transaction[_SYMBOL], str(transaction[_AMOUNT]), str(transaction[_COST]))
+        if transaction[_SIDE] == _BUY:
+            out_asset = trade.quote_asset
+            in_asset = trade.base_asset
+            crypto_in = RP2Decimal(str(transaction[_AMOUNT]))
+            conversion_info = f"{trade.quote_info} -> {trade.base_info}"
+        elif transaction[_SIDE] == _SELL:
+            out_asset = trade.base_asset
+            in_asset = trade.quote_asset
+            crypto_in = RP2Decimal(str(transaction[_COST]))
+            conversion_info = f"{trade.base_info} -> {trade.quote_info}"
+        else:
+            raise Exception(f"Internal error: unrecognized transaction side: {transaction[_SIDE]}")
+
+        if fee_asset == in_asset:
+            crypto_fee = RP2Decimal(str(transaction[_FEE][_COST]))
+        else:
+            crypto_fee = ZERO
+            transaction_fee = RP2Decimal(transaction[_FEE][_COST])
```

Comment:
Is transaction[_FEE][_COST] a float? If so the right way to convert it is: `RP2Decimal(str(transaction[_FEE][_COST]))`